src/estimators/any_sketch.py

```diff
@@ -47,6 +47,54 @@ def __call__(self, x, y):
     return x | y
 
 
+class UniqueKeyFunction(ValueFunction):
+  """ValueFunction to track the state of unique key of a register."""
+
+  FLAG_EMPTY_REGISTER = 0
+  FLAG_COLLIDED_REGISTER = -1
+
+  def __call__(self, x, y):
+    """ValueFunction to track the state of unique key of a register.
+
+    Args:
+      x: A state of unique key. It can be either a real key (hashed ID)
+        indicating the unique key in the register, or FLAG_EMPTY_REGISTER
+        indicating that the register is empty, or FLAG_COLLIDED_REGISTER
+        indicating that the register already has collision.
+      y: Another state of unique key.
+
+    Returns:
+      A state of unique key that merges x and y.
+    """
+    if x == UniqueKeyFunction.FLAG_EMPTY_REGISTER:
+      return y
+    if y == UniqueKeyFunction.FLAG_EMPTY_REGISTER:
+      return x
+    if x == UniqueKeyFunction.FLAG_COLLIDED_REGISTER or y == UniqueKeyFunction.FLAG_COLLIDED_REGISTER:
+      return UniqueKeyFunction.FLAG_COLLIDED_REGISTER
+    # When x & y are neither FLAG_EMPTY_REGISTER or FLAG_COLLIDED_REGISTER,
+    # i.e., they are real keys,
+    # it suffices to check if they are the same keys.
+    if x == y:
+      return x
+    return UniqueKeyFunction.FLAG_COLLIDED_REGISTER
```

Comment:
Sounds the logic can be simplified? Considering it's `FLAG_EMPTY_REGISTER` only if both x and y == FLAG_EMPTY_REGISTER
```suggestion
    if x == y:
      return x
    return UniqueKeyFunction.FLAG_COLLIDED_REGISTER
```

Reply:
Thanks! Please see the revision with simplified logic and additional comments.